src/scenes/ObservationsRouter/BiomassMeasurementMapView.tsx

```diff
@@ -0,0 +1,226 @@
+  }, [observationsResults]);
+
+  const [selectedObservationDate, setSelectedObservationDate] = useState<string | undefined>();
+
+  useEffect(() => {
+    if (observationHistory?.status === 'success') {
+      setPlantingSiteHistory(observationHistory.data);
+    }
+  }, [observationHistory]);
+
+  useEffect(() => {
+    if (observationsDates) {
+      setSelectedObservationDate((currentDate) => {
+        if ((!currentDate || !observationsDates.includes(currentDate)) && observationsDates.length > 0) {
+          return observationsDates[observationsDates.length - 1];
+        } else {
+          return currentDate;
+        }
+      });
+    } else {
+      setSelectedObservationDate('');
+    }
+  }, [observationsDates]);
+
+  const selectedObservation = useMemo(
+    () =>
+      observationsResults?.find((obs) => {
+        const timeZone = selectedPlantingSite?.timeZone ?? defaultTimeZone.get().id;
+        const dateToCheck = obs.completedTime ? getDateDisplayValue(obs.completedTime, timeZone) : obs.startDate;
+        return dateToCheck === selectedObservationDate;
+      }),
+    [observationsResults, selectedObservationDate]
+  );
+
+  const observationData = useMemo(() => {
+    return observations.find((obv) => obv.id === selectedObservation?.observationId);
+  }, [observations, selectedObservation]);
+
+  useEffect(() => {
+    if (observationData) {
+      const historyId = observationData.plantingSiteHistoryId;
+      if (historyId) {
+        const requestObservationHistory = dispatch(
+          requestGetPlantingSiteHistory({
+            plantingSiteId: selectedPlantingSite.id,
+            historyId: historyId,
+          })
+        );
+        setRequestId(requestObservationHistory.requestId);
+      }
+    } else {
+      setRequestId('');
+    }
+  }, [observationData]);
+
+  const plantingSiteMapData: MapSourceBaseData | undefined = useMemo(
+    () => MapService.getMapDataFromPlantingSite(selectedPlantingSite)?.site,
+    [selectedPlantingSite]
+  );
+
+  const mapData: Record<MapObject, MapSourceBaseData | undefined> = useMemo(() => {
+    if (!selectedObservationDate || !selectedObservation) {
+      return {
+        site: plantingSiteMapData,
+        zone: undefined,
+        subzone: undefined,
+        permanentPlot: undefined,
+        temporaryPlot: undefined,
+        adHocPlot: undefined,
+      };
+    }
+
+    return MapService.getMapDataFromObservation(selectedObservation, plantingSiteHistory);
+  }, [selectedObservation, selectedObservationDate, plantingSiteMapData, plantingSiteHistory]);
+
+  const layerOptions: MapLayer[] = ['Planting Site', 'Zones', 'Monitoring Plots'];
+  const [includedLayers, setIncludedLayers] = useState<MapLayer[]>(layerOptions);
+
+  const layerOptionLabels: Record<MapLayer, string> = {
+    'Planting Site': strings.PLANTING_SITE,
+    Zones: strings.ZONES,
+    'Sub-Zones': strings.SUBZONES,
+    'Monitoring Plots': strings.MONITORING_PLOTS,
+  };
```

Comment:
`layerOptions` is missing one of the layers from `layerOptionLabels`: `Sub-Zones`. Is this intentional?

Reply:
We are not showing subzones in observations maps, the label will not show since layer options does not include it. The label could be removed